sdk/action.py

```diff
@@ -0,0 +1,174 @@
+    def __init__(self):
+        err = "Action_Type_Error"
+        super().__init__(self, err)
+
+def get_action_from_abi_json(action, abi_json):
+    libevt.init_lib()
+    abi_dict = json.loads(abi_json)
+    _bin = abi.json_to_bin(action, abi_json).to_hex_string()
+    if action == "newdomain":
+        return NewDomainAction(abi_dict["name"], _bin)
+    elif action == "updatedomain":
+        return UpdateDomainAction(abi_dict["name"], _bin)
+    elif action == "newgroup":
+        return NewGroupAction(abi_dict["name"], _bin)
+    elif action == "updategroup":
+        return UpdateGroupAction(abi_dict["name"], _bin)
+    elif action == "newaccount":
+        return NewAccountAction(abi_dict["name"], _bin)
+    elif action == "updateowner":
+        return UpdateOwnerAction(abi_dict["name"], _bin)
+    elif action == "transferevt":
+        return TransferEvtAction(abi_dict["from"], _bin)
+    elif action == "issuetoken":
+        return IssueTokenAction(abi_dict["domain"], _bin)
+    elif action == "transfer":
+        return TransferAction(abi_dict["domain"], abi_dict["name"], _bin)
+    else:
+        return None
+
+class ActionGenerator:
+    def newdomain(self, name, issuer, **kwargs):
+        auth_A = base.AuthorizerWeight(base.AuthorizerRef("A", issuer), 1)
+        auth_G = base.AuthorizerWeight(base.AuthorizerRef("G", "OWNER"), 1)
+        issue = kwargs["issue"] if "issue" in kwargs else None
+        transfer = kwargs["transfer"] if "transfer" in kwargs else None
+        manage = kwargs["manage"] if "manage" in kwargs else None
+        abi_json = base.NewDomainAbi(name=name,
+                issuer=str(issuer),
+                issue=issue or base.PermissionDef("issue", 1, [auth_A]),
+                transfer=transfer or base.PermissionDef("transfer", 1, [auth_G]),
+                manage=manage or base.PermissionDef("manage", 1, [auth_A]))
+        return get_action_from_abi_json("newdomain", abi_json.dumps())
+
+    def updatedomain(self, name, issuer, **kwargs):
+        auth_A = base.AuthorizerWeight(base.AuthorizerRef("A", issuer), 1)
+        auth_G = base.AuthorizerWeight(base.AuthorizerRef("G", "OWNER"), 1)
+        issue = kwargs["issue"] if "issue" in kwargs else None
+        transfer = kwargs["transfer"] if "transfer" in kwargs else None
+        manage = kwargs["manage"] if "manage" in kwargs else None
+        abi_json = base.NewDomainAbi(name=name,
+                issuer=str(issuer),
+                issue=issue or base.PermissionDef("issue", 1, [auth_A]),
+                transfer=transfer or base.PermissionDef("transfer", 1, [auth_G]),
+                manage=manage or base.PermissionDef("manage", 1, [auth_A]))
+        return get_action_from_abi_json("updatedomain", abi_json.dumps())
+
+    def newgroup(self, name, key, **kwargs):
+        group = kwargs["group"] if "group" in kwargs else None
+        if group == None:
+            pub1, priv1 = ecc.generate_new_pair()

```

Comment:
for new group, don't put example structure here, require user a json of group or `base.Group` object.